commit cf45505bc7713f344a4b51b64fb0a64c20b2c04f
Author:     Oliver Breitwieser <oliver@breitwieser.eu>
AuthorDate: 2025-11-11 21:51:49 +0100
Commit:     Oliver Breitwieser <oliver@breitwieser.eu>
CommitDate: 2026-08-23 15:13:27 +0200

    tmux-fzf: unstable-2023-10-24 -> 2025-09-24

diff --git a/pkgs/misc/tmux-plugins/default.nix b/pkgs/misc/tmux-plugins/default.nix
index 4bfefe112979..e2157af86c19 100644
--- a/pkgs/misc/tmux-plugins/default.nix
+++ b/pkgs/misc/tmux-plugins/default.nix
@@ -1036,24 +1036,38 @@ in
       rev = "05af76daa2487575b93a4f604693b00969f19c2f";
       hash = "sha256-ay7z0MkeDCpxdwNTKFrkxi/hUE7a5K7P7oFhfn94aLA=";
     };
+
+    # NOTE: not patching calls to copyq as these would introduce a fixed dependency
     postInstall = ''
-      find $target -type f -print0 | xargs -0 sed -i -e 's|fzf |${pkgs.fzf}/bin/fzf |g'
-      find $target -type f -print0 | xargs -0 sed -i -e 's|sed |${pkgs.gnused}/bin/sed |g'
-      find $target -type f -print0 | xargs -0 sed -i -e 's|tput |${pkgs.ncurses}/bin/tput |g'
+      # always called with arguments
+      find $target -type f -print0 | xargs -0 sed -i -e 's|\bawk\b |${pkgs.gawk}/bin/awk |g'
+      find $target -type f -print0 | xargs -0 sed -i -e 's|\bfzf\b |${pkgs.fzf}/bin/fzf |g'
+      find $target -type f -print0 | xargs -0 sed -i -e 's|\bkill\b |${pkgs.coreutils}/bin/kill |g'
+      find $target -type f -print0 | xargs -0 sed -i -e 's|\bps\b |${pkgs.unixtools.ps}/bin/ps |g'
+      find $target -type f -print0 | xargs -0 sed -i -e 's|\bpstree\b |${pkgs.pstree}/bin/pstree |g'
+      find $target -type f -print0 | xargs -0 sed -i -e 's|\bsed\b |${pkgs.gnused}/bin/sed |g'
+      find $target -type f -print0 | xargs -0 sed -i -e 's|\btop\b |${pkgs.unixtools.top}/bin/top |g'
+      find $target -type f -print0 | xargs -0 sed -i -e 's|\btput\b |${pkgs.ncurses}/bin/tput |g'
+
+      # called as single commands in sub-shells
+      find $target -type f -print0 | xargs -0 sed -i -e 's|\buname\b|${pkgs.coreutils}/bin/uname|g'
+      find $target -type f -print0 | xargs -0 sed -i -e 's|\bwhoami\b|${pkgs.coreutils}/bin/whoami|g'
     '';
     meta = {
       homepage = "https://github.com/sainnhe/tmux-fzf";
       description = "Use fzf to manage your tmux work environment! ";
       longDescription = ''
         Features:
-        * Manage sessions (attach, detach*, rename, kill*).
-        * Manage windows (switch, link, move, swap, rename, kill*).
-        * Manage panes (switch, break, join*, swap, layout, kill*, resize).
-        * Multiple selection (support for actions marked by *).
+        * Manage sessions (switch, new, rename, detach, kill).
+        * Manage windows (switch, link, move, swap, rename, kill).
+        * Manage panes (switch, break, join, swap, layout, kill, resize).
         * Search commands and append to command prompt.
         * Search key bindings and execute.
-        * User menu.
-        * Popup window support.
+        * Search clipboard history and paste to current window.
+        * Process management (top, pstree, terminate, kill, interrupt, continue, stop, quit, hangup).
+        * User menu (run custom commands).
+        * Preview sessions, windows and panes.
+        * Multiple selection.
       '';
       license = lib.licenses.mit;
       platforms = lib.platforms.unix;
