--- b/pkgs/development/compilers/crystal/default.nix
+++ a/pkgs/development/compilers/crystal/default.nix
@@ -19,7 +19,6 @@
   llvmPackages_19,
   llvmPackages_20,
   llvmPackages_21,
-  llvmPackages_22,
   makeWrapper,
   openssl,
   pcre2,
@@ -333,13 +332,5 @@
     doCheck = false;
   };
 
+  crystal = crystal_1_18;
-  crystal_1_19 = generic {
-    version = "1.19.1";
-    sha256 = "sha256-vMS2GJb6c6RvflDSS2EWHsERJ0rvzZMVm50gaTXRs4Y=";
-    binary = binaryCrystal_1_10;
-    llvmPackages = llvmPackages_22;
-    doCheck = false;
-  };
-
-  crystal = crystal_1_19;
 }
--- b/pkgs/top-level/all-packages.nix
+++ a/pkgs/top-level/all-packages.nix
@@ -4033,7 +4033,6 @@
     crystal_1_16
     crystal_1_17
     crystal_1_18
-    crystal_1_19
     crystal
     ;
 
